src/components/Contact.js

```diff
@@ -3,10 +3,10 @@ import { Form } from 'semantic-ui-react';
 
 class Contact extends Component {
     default = { firstName: '', lastName: '', email: '', phone: '' };
-    state = { ...this.default }; 
-  
-    setValue = (e) => {
-        let { target: { id, value }} = e;
+    state = { ...this.default };
+
+    setValue = ({ target: { id, value }}) => {
+      this.setState({[id]: value});
     }
```

Comment:
`setValue()` is the function being passed into each input field in the form below. Specifically, this is begin passed into the `onChange` prop of the `<Form.Field>` component. This is exactly what we discussed yesterday. `<Form.Field>` is a react component and in the code below, we're passing a function to the 'onChange' prop. Behind the scenes, the Form.Field component creates pass the value of the `onChange` prop and sets it the underlying `onChange` event that is [native to HTML inputs](https://www.w3schools.com/jsref/event_onchange.asp). 

How whenever something 'changes' aka a user types in the input, your `setValue` function is called. The underlying `onChange` event has event information that is passed back to that function. That's why this function originally had an  `e` as it's argument here: 

```
 setValue = (e) => {}
```
this is pass 'event' is just an object being passed back with a whole bunch of information. But at the core, it's just an object that looks like this: `e = {target: {id: 'componentDefinedID', value: 'whateverIsTheValueOfTheField'}, <a bunch of other interesting but not relevant stuff>}` 

Ok...so now we have the typed values coming back to this `setValue()` function. As we discussed yesterday, every component has a `state`. This is just a place to store data that will exist as long as the component is rendered. The other nice thing about React state, is every time you make a change to it, it will compare the impacts of that change in the Virtual DOM and the actual DOM and 're-render' the component as need. 

Since each of the form input fields in the Contact Form are each connected with the state of their respective name (ie state = {firstName: ''} and the input with the id of firstName is connected to it), we need to update the state with this new value. That's where [`this.setState()`](https://facebook.github.io/react/docs/react-component.html#setstate) comes in. It accepts and object and will update the component state and rerender the dom as needed. For example: `this.setState({firstName: 'Josh'})`

I've done just that inside of `setValue()` by passing the 'id' (or the field type) in as the key and the value of what the person has typed as the value. 

```
this.setState({[id]: value});
```